src/site/stages/build/plugins/update-robots.js

```diff
@@ -0,0 +1,26 @@
+// Dependencies
+const ENVIRONMENTS = require('../../../constants/environments');
+
+function updateRobots(buildOptions) {
+  // Use the default robots.txt file on production.
+  if (buildOptions.buildtype === ENVIRONMENTS.VAGOVPROD) {
+    // eslint-disable-next-line
+    console.log('Using the production robots.txt');
+    return () => {};
+  }
+
+  return (files, smith, done) => {
+    // Derive the robots file.
+    const robots = files['robots.txt'];
+
+    // Update the robots.txt contents to disallow crawlers.
+    robots.contents = new Buffer('User-agent: *\nDisallow: /\n');
+
+    // eslint-disable-next-line
+    console.log('Using the non-production robots.txt');
+
+    done();
```

Comment:
Alternatively, your function could look like:
```js
return files => {
  ...
}
```
Through a series of JavaScript black magic rituals, Metalsmith lets you omit the `done` argument if it's not needed. Since you're neither doing anything asynchronous here nor returning early, it's not strictly speaking needed.

See the [filenames plugin](https://github.com/MoOx/metalsmith-filenames/blob/master/src/index.es) for an example.

Reply:
Great suggestion! Updated! :)